### DISCUSSION TITLE
Next.js Redirects

### DISCUSSION THREAD

**Discussion by allindeveloper** [vercel/next.js#53386] [Help]
### Summary

I am trying to implement a custom redirections in nextjs using `redirects` in `next.config.js` but it doesn't seem to work.

Or does someone have a different solution to solve this

### Additional information
This works fine

```js

  async redirects() {
    return [
       {
        source: '/:store-:locale/:slug*',
        destination: '/:de/:slug*', // fallback to only de
        permanent: false,
      },
    ];
  },
```

But this not work.
Expectation: I expected next.js to pick locale path from the url and replace in the destination and route to it

```js
  async redirects() {
    return [
       {
        source: '/:store-:locale/:slug*',
        destination: '/:locale/:slug*',  // fallback to locale matched and picked from url
        permanent: false,
      },
    ];
  },
```
```


### Example

_No response_

**Comment by Nadeera3784** [comment]
It seems like you want to implement a custom redirection mechanism in Next.js using the `next.config.js` file. The purpose is to extract the `:locale` parameter from the source URL and use it in the destination URL for redirection. However, you've encountered an issue with the redirection not working as expected.

Your expectation is to have Next.js pick the `:locale` parameter from the source URL and replace it in the destination URL for redirection.

To achieve this, you can make use of the `rewrites` property in the `next.config.js` file instead of `redirects`. The `rewrites` property allows you to modify the incoming URL before it is matched to a page.

Here's how you can implement it:

```javascript
module.exports = {
  async rewrites() {
    return [
      {
        source: '/:store-:locale/:slug*',
        destination: '/:locale/:slug*', // Replace :locale with the matched value from the source URL
      },
    ];
  },
};
```

In this configuration, `:store` will be ignored, and `:locale` will be extracted from the source URL and used in the destination URL for redirection.

Remember that for this to work, you should ensure that you have your page structure set up to handle the `:locale` and `:slug` parameters correctly. If your pages or components depend on the `:store` parameter, you might need to handle it accordingly in your pages or components.

**Reply by allindeveloper** [reply]
Thank you @Nadeera3784  
this does not update the url on the browser, it just masks the destination. so i don't think this would work effectively.

**Comment by allindeveloper** [comment]
I ended up using the same redirects and it worked.

**Reply by Nadeera3784** [reply]
Oh glad to hear that @allindeveloper
